src/main/java/com/gaggle/contact/controllers/ContactController.java

```diff
@@ -44,6 +46,20 @@
                 .orElseThrow(() -> new ContactNotFoundException(name));
     }
 
+    @GetMapping("/query")
+    public List<Contact> getByEmail(@RequestParam final String email) {
+        if (Objects.isNull(email) || !email.contains("@")) {
```

Comment:
These lines can be omitted and Spring can validate the params by annotating as required and Spring recognizes email as a type and will validate it.